src/ovs/aca_ovs_l3_programmer.cpp

```diff
@@ -42,49 +42,346 @@ ACA_OVS_L3_Programmer &ACA_OVS_L3_Programmer::get_instance()
+          current_gateway_mac = found_gateway_mac;
+          current_gateway_mac.erase(
+                  remove(current_gateway_mac.begin(), current_gateway_mac.end(), ':'),
+                  current_gateway_mac.end());
+
+          addr = inet_network(found_gateway_ip.c_str());
+          snprintf(hex_ip_buffer, HEX_IP_BUFFER_SIZE, "0x%08x", addr);
+
+          // Program ARP responder:
+          cmd_string = "add-flow br-tun \"table=51,priority=50,arp,dl_vlan=" +
+                       to_string(source_vlan_id) + ",nw_dst=" + found_gateway_ip +
+                       " actions=move:NXM_OF_ETH_SRC[]->NXM_OF_ETH_DST[],mod_dl_src:" + found_gateway_mac +
+                       ",load:0x2->NXM_OF_ARP_OP[],move:NXM_NX_ARP_SHA[]->NXM_NX_ARP_THA[],move:NXM_OF_ARP_SPA[]->NXM_OF_ARP_TPA[],load:0x" +
+                       current_gateway_mac +
+                       "->NXM_NX_ARP_SHA[],load:" + string(hex_ip_buffer) +
+                       "->NXM_OF_ARP_SPA[],in_port\"";
+
+          ACA_OVS_L2_Programmer::get_instance().execute_openflow_command(
+                  cmd_string, dataplane_programming_time, overall_rc);
+
+          // Program ICMP responder:
+          cmd_string =
+                  "add-flow br-tun \"table=52,priority=50,icmp,dl_vlan=" +
+                  to_string(source_vlan_id) + ",nw_dst=" + found_gateway_ip +
+                  " actions=move:NXM_OF_ETH_SRC[]->NXM_OF_ETH_DST[],mod_dl_src:" + found_gateway_mac +
+                  ",move:NXM_OF_IP_SRC[]->NXM_OF_IP_DST[],mod_nw_src:" + found_gateway_ip +
+                  ",load:0xff->NXM_NX_IP_TTL[],load:0->NXM_OF_ICMP_TYPE[],in_port\"";
+
+          ACA_OVS_L2_Programmer::get_instance().execute_openflow_command(
+                  cmd_string, dataplane_programming_time, overall_rc);
+
+          // Should be able to ping the gateway now
+
+          for (int k = 0; k < current_subnet_routing_table.routing_rules_size(); k++) {
+            auto current_routing_rule = current_subnet_routing_table.routing_rules(k);
+
+            // check if current_routing_rule already exist in new_subnet_routing_tables
+            if (new_subnet_routing_table_entry.routing_rules.find(
+                        current_routing_rule.id()) !=
+                new_subnet_routing_table_entry.routing_rules.end()) {
+              is_routing_rule_exist = true;
+            }
+
+            // populate the routing_rule_table_entry and add that to
+            // new_subnet_routing_table_entry.routing_rules only if the
+            // operation type for that routing_rule is CREATE/UPDATE/INFO
+            if ((current_routing_rule.operation_type() == OperationType::CREATE) ||
+                (current_routing_rule.operation_type() == OperationType::UPDATE) ||
+                (current_routing_rule.operation_type() == OperationType::INFO)) {
+              routing_rule_table_entry routing_rule_table_entry;
+
+              if (is_routing_rule_exist) {
+                routing_rule_table_entry =
```

Comment:
should we rename it to routing_rule_entry, as we have subnet_routing_table_entry above?

Reply:
changed, that will map nicely with our schema.